SignalServiceKit/src/Util/OWSAnalytics.m

```diff
@@ -157,6 +157,12 @@ - (void)tryToSyncEvents
 
         DDLogDebug(@"%@ trying to deliver event: %@", self.tag, firstEventKey);
         self.hasRequestInFlight = YES;
+
+        __block UIBackgroundTaskIdentifier task;
+        task = [UIApplication.sharedApplication beginBackgroundTaskWithExpirationHandler:^{
+            [UIApplication.sharedApplication endBackgroundTask:task];
```

Comment:
do we need to reset `hasRequestInFlight = NO` in our expiration handler? Any other state?